src/diagnostic-channel-publishers/src/azure-coretracing.pub.ts

```diff
@@ -57,18 +57,20 @@ const azureCoreTracingPatchFunction: PatchFunction = (coreTracing: typeof coreTr
                 const getTracerOriginal = tracerProvider.getTracer;
                 tracerProvider.getTracer = function(tracerName, version) {
                     const tracer = getTracerOriginal.call(this, tracerName, version);
-                    const startSpanOriginal = tracer.startSpan;
-                    tracer.startSpan = function(spanName: string, options?: opentelemetryTypes.SpanOptions, context?: opentelemetryTypes.Context) {
-                        const span = startSpanOriginal.call(this, spanName, options, context);
-                        const originalEnd = span.end;
-                        span.end = function() {
-                            const result = originalEnd.apply(this, arguments);
-                            channel.publish("azure-coretracing", span);
-                            return result;
+                    if (!tracer[exports.AzureMonitorSymbol]) { // Avoid patching multiple times
+                        const startSpanOriginal = tracer.startSpan;
+                        tracer.startSpan = function(spanName: string, options?: opentelemetryTypes.SpanOptions, context?: opentelemetryTypes.Context) {
+                            const span = startSpanOriginal.call(this, spanName, options, context);
+                            const originalEnd = span.end;
+                            span.end = function() {
+                                const result = originalEnd.apply(this, arguments);
+                                channel.publish("azure-coretracing", span);
+                                return result;
+                            };
+                            return span;
                         };
-                        return span;
-                    };
-                    tracer[AzureMonitorSymbol] = true;
+                        tracer[AzureMonitorSymbol] = true;
```

Comment:
Note: There are some runtime environments that will freeze objects and that will cause this line to throw 😢 
AI Web SDK just ignores this type of failure as there is no easy workaround
https://github.com/microsoft/ApplicationInsights-JS/blob/master/channels/applicationinsights-channel-js/src/Sender.ts#L625-L630

Reply:
And for ajax tracking it just disables tracking for this case https://github.com/microsoft/ApplicationInsights-JS/blob/master/extensions/applicationinsights-dependencies-js/src/ajax.ts#L64-L81